### PR TITLE
Initial release of setup-toit action

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,20 @@
...
+
+## Basic usage
+
+See [action.yml](action.yml) for a complete list of options and outputs.
+
+```yaml
+steps:
+- uses: actions/checkout@v4
+
+- uses: toitware/action-setup@v1
```
Comment (kasperl) [inline README.md]:
```suggestion
- uses: toitlang/action-setup@v1
```

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        echo "TOIT_INSTALL_DIR=$INSTALL_DIR" >> $GITHUB_ENV
+
+    - name: Create temporary directory
+      # Windows.
+      if: startsWith(runner.os, 'Windows')
+      shell: powershell
+      run: |
+        $tempDir = New-TemporaryFile | % { $_.DirectoryName }
+        echo "TOIT_INSTALL_DIR=$tempDir" | Out-File -FilePath $env:GITHUB_ENV -Append
+        echo "Temp directory is $tempDir"
```
Comment (kasperl) [inline action.yml]:
```suggestion
        echo "Temporary directory is $tempDir"
```

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    - name: Output tmp-dir
+      shell: bash
+      id: tmp-dir
+      run: |
+        echo TOIT_INSTALL_DIR=$TOIT_INSTALL_DIR >> $GITHUB_OUTPUT
+
```
Comment (kasperl) [inline action.yml:67]: Drop one newline.

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    - name: Create temporary directory
+      # Windows.
+      if: startsWith(runner.os, 'Windows')
+      shell: powershell
+      run: |
+        $tempDir = New-TemporaryFile | % { $_.DirectoryName }
+        echo "TOIT_INSTALL_DIR=$tempDir" | Out-File -FilePath $env:GITHUB_ENV -Append
+        echo "Temp directory is $tempDir"
+
+    - name: Output tmp-dir
```
Comment (kasperl) [inline action.yml]:
```suggestion
    - name: Output temporary directory
```

File: action.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        target: ${{ steps.tmp-dir.outputs.TOIT_INSTALL_DIR }}
+
+    - name: Extract Toit
+      shell: bash
+      run: |
+        cd '${{ steps.tmp-dir.outputs.TOIT_INSTALL_DIR }}'
+        # Extracts the tar.gz leaving a 'toit' directory containing 'bin', 'tools' and 'lib' directory.
+        tar x -zf toit-*.tar.gz
+
+    - name: Add to Path
```
Comment (kasperl) [inline action.yml]:
```suggestion
    - name: Add to path
```

or perhaps:

```suggestion
    - name: Add to PATH
```

File: action.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+      # Linux and macOS.
+      if : startsWith(runner.os, 'Linux') || startsWith(runner.os, 'macOS')
+      shell: bash
+      run: |
+        echo "${{ steps.tmp-dir.outputs.TOIT_INSTALL_DIR }}/toit/bin" >> $GITHUB_PATH
+
+    - name: Add to Path
```
Comment (kasperl) [inline action.yml]:
```suggestion
    - name: Add to path
```

or perhaps:

```suggestion
    - name: Add to PATH
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: action.yml (hunk 3 above)
Comment (floitsch) [reply action.yml:67]: done.

**Pull request merged**